package.json

```diff
@@ -11,13 +11,17 @@
     "ajv-keywords": "^2.0.0",
     "async": "^2.1.2",
     "chrome-trace-event": "^0.1.1",
+    "css-selector-tokenizer": "^0.7.0",
     "enhanced-resolve": "^4.0.0-beta.2",
     "eslint-scope": "^3.7.1",
+    "generic-names": "^1.0.3",
     "loader-runner": "^2.3.0",
     "loader-utils": "^1.1.0",
     "memory-fs": "~0.4.1",
     "mkdirp": "~0.5.0",
     "node-libs-browser": "^2.0.0",
+    "postcss": "^6.0.16",
+    "postcss-value-parser": "^3.3.0",
```

Comment:
Better for use https://github.com/shellscape/postcss-values-parser, we use this package in prettier, also stylelint in near future want to switch on this, also it is package @shellscape and we can fix some problems asasp

Reply:
@evilebottnawi You can fix some problems with value-parser too. I'm open to major releases.

Reply:
@TrySound I'm not against you packages, I just want to express my opinion, we have very few contributors, although these are quite vital packages for many tools, ideally I would see the merging of packages into one, if possible (maybe options for backward compatibility). This would be a great solution, also perhaps we could bring his to stable when they became part of postcss in next major release. Just my opinion. What do you think about this?

Reply:
I'd say there's done a lot of work on this code and changing dependency just because it's nice is wasting of time. postcss just considered adding value parser out of the box. You can't be sure it will be 1:1 like postcss-values-parser and rely on it. Let's rewrite when there will be at least postcss beta or similar.

Reply:
Guys I'm just had't the time to finalise the local stuff I'm working on for `postcss-icss-*` side of affiars, this stuff (dependencies) will be moved soon from the PR :D 😛. Happy with every `postcss-value-parser` that is capable of parsing `@import/url()` for the `CSSParser` in it's default state. `css-modules` will be an opt-in either via `postcss-loader` or something like

```js
{
  type: 'css/experimental',
  test: /\.css$/,
  parser : {
    plugins: [
       require('postcss-modules')(options)
    ]
  }
}
```

Reply:
@michael-ciniawsky Btw, I think you can get rid from a lot of problems if you will still preserve composes/selectors/values order and will even keep 'em in one package. Keeping them separated was my fall.

Reply:
Oh, you didn't add composes and values. Well, I like it even more. Those features are hard to maintain.

Reply:
@TrySound 🤔 Something like 

```js
const { imports, urls } = require('@cssm/esm')
const { selectors, composes, keyframes, values } = require('@cssm/modules')
```

would be awesome

Reply:
```
js
const modules = require('@cssm/modules')

postcss([
   modules({
      values: true || false || ...options
      selectors: true || false || ...options
      ...
   })
])
```

as a Wrapper

Reply:
@michael-ciniawsky Yeah, but as I said order is important too and would be good to maintain second case in one pack
```js
cssm({ selectors: true, composes: true, values: true })
```

Reply:
I wouldn't even add options per each plugin.

Reply:
The `@cssm` would scope is available, but that's minor important atm :)

So you suggesting to move `postcss-icss-selctors,values,composes,keyframes` into one repo/package, handle intermediate 'state' within the `modules` plugin itself (no messages) and let it just output the (webpack) dependencies (e.g `export const selector = 'name-2358'` etc) via `result.messages` ?

Reply:
Yep. Because maintaining dependent stuff with different order is shooting into feet.

Reply:
Should we do it in `postcss-icss-selectors` ? (rename, move the other plugins in there) and discuss the rest there then ? I get my things sorted and open an PR :)

`postcss-icss-{url, import}` also into one package? I would personally say yes and did it for HTML https://github.com/posthtml/esm :)

Reply:
Yeah. I think it would be even better. A trend of "big" packages became popular again :)

Reply:
Please not anything 'monorepo-ish'  if avoidable :D, maybe for `@cssm/modules` but a `lib/` with a few files and a `index.js` also does the job and is idiot proofen :joy:

BTW anyone interested can try this branch out here(https://github.com/michael-ciniawsky/webpack-css). Pulls this PR from github with a small boilerplate test setup to tinker around :)